traffic_ops/testing/api/v4/roles_test.go

```diff
@@ -63,30 +62,23 @@ func UpdateTestRolesWithHeaders(t *testing.T, header http.Header) {
 		t.Fatal("Need at least one Role to test updating a Role with HTTP headers")
 	}
 	firstRole := testData.Roles[0]
-	if firstRole.Name == nil {
-		t.Fatal("Found a Role in the testing data with null or undefined name")
-	}
 
 	// Retrieve the Role by role so we can get the id for the Update
 	opts := client.NewRequestOptions()
 	opts.Header = header
-	opts.QueryParameters.Set("name", *firstRole.Name)
+	opts.QueryParameters.Set("name", firstRole.Name)
 	resp, _, err := TOSession.GetRoles(opts)
 	if err != nil {
-		t.Errorf("cannot get Role '%s' by name: %v - alerts: %+v", *firstRole.Name, err, resp.Alerts)
+		t.Errorf("cannot get Role '%s' by name: %v - alerts: %+v", firstRole.Name, err, resp.Alerts)
 	}
 	if len(resp.Response) != 1 {
-		t.Fatalf("Expected exactly one Role to exist with name '%s', found: %d", *firstRole.Name, len(resp.Response))
+		t.Fatalf("Expected exactly one Role to exist with name '%s', found: %d", firstRole.Name, len(resp.Response))
 	}
 	remoteRole := resp.Response[0]
-	if remoteRole.ID == nil {
-		t.Fatal("Traffic Ops returned a representation for a Role with null or undefined ID")
-	}
-
-	expectedRole := "new admin2"
-	remoteRole.Name = &expectedRole
+	expectedDescription := "new description"
+	remoteRole.Description = expectedDescription
 	opts.QueryParameters.Del("name")
-	_, reqInf, _ := TOSession.UpdateRole(*remoteRole.ID, remoteRole, opts)
+	_, reqInf, err := TOSession.UpdateRole(remoteRole.Name, remoteRole, opts)
```

Comment:
you're now capturing the error, but the value is never read after this assignment